crates/ruff/src/rules/pandas_vet/rules/inplace_argument.rs

```diff
@@ -77,9 +77,11 @@ pub fn inplace_argument(
                 //    the star argument _doesn't_ contain an override).
                 // 2. The call is part of a larger expression (we're converting an expression to a
                 //    statement, and expressions can't contain statements).
+                // 3. The call is in a lambda (we can't assign to a variable in a lambda).
                 let fixable = !seen_star
                     && matches!(checker.ctx.current_stmt().node, StmtKind::Expr { .. })
-                    && checker.ctx.current_expr_parent().is_none();
+                    && checker.ctx.current_expr_parent().is_none()
+                    && !checker.ctx.scope().kind.is_lambda();
```

Comment:
Ah yeah, the existing checks actually _should_ catch this, but it has to do with a limitation in the context we "restore" when parsing deferred bodies (like lambda bodies). In short, we don't restore the expression stack, since it's expensive to track all expressions, and I hadn't realized it was necessary, but it is here.

Reply:
I'll add a TODO and merge.

Reply:
Ah right. I wondered why (2) was not catching it but didn't have time to dig deeper 😅